connector/web3/web3.go

```diff
@@ -68,6 +83,42 @@ func (c *web3Connector) Verify(address, msg, signedMsg string) (identity connect
 	return identity, nil
 }
 
+func (c *web3Connector) VerifyERC1271Signature(contractAddress common.Address, hash, signature []byte) (identity connector.Identity, err error) {
+	if c.ethClient == nil {
+		c.logger.Errorf("Eth client was not initialized successfully %v", err)
+		return identity, errors.New("error occurred completing authentication, please try again")
+	}
+	var msgHash [32]byte
+	copy(msgHash[:], hash)
+
+	// ContractLogin is just a simple interface with the signature for IsValidSignature
+	/**
+	 * function isValidSignature(bytes32 hash, bytes memory signature) external view returns (bytes4 magicValue);
+	 */
+	ct, err := NewContractLogin(contractAddress, c.ethClient)
+	if err != nil {
+		return identity, errors.New("error occurred completing login")
+	}
+
+	isValid, err := ct.IsValidSignature(&bind.CallOpts{
+		Pending: false,
+	}, msgHash, signature)
+	if err != nil {
+		return identity, fmt.Errorf("error occurred completing login %w", err)
+	}
+	resultVal := common.BytesToAddress(isValid[:])
```

Comment:
Address? That doesn't seem right. Isn't it four bytes? You're doing `bytes.Equal` later, do you need to convert this at all?

Reply:
The response is 4 bytes, it was difficult working with it, especially comparing it to []byte("0xffffffff"). So i decided to pad it and make both 20bytes which makes it easier to compare.

Reply:
Reiterating: 0xffffffff is not a value you want to use in this code. It's not the "magic value" from the ERC.

Reply:
Remember the existing of `common.FromHex("0xffffffff")`, also (but again, you don't want to use this value).

Reply:
Let's not do this address thing! We have two `[]byte` let's just call `bytes.Equal`!